front/lib/stores/appStore.tsx

```diff
@@ -0,0 +1,186 @@
+import { del, get, set } from "idb-keyval";
+import { createStore } from "zustand";
+import { createJSONStorage, devtools, persist } from "zustand/middleware";
+
+import type { LightAgentConfigurationType } from "@app/types";
+
+export interface AppStoreInitialState {
+  userId?: string;
+  workspaceId?: string;
+  timestamp: number | null;
+  actions: {
+    setUserId: (userId: string) => void;
+    getAgentConfigurationsCache: ({
+      userId,
+      workspaceId,
+    }: {
+      userId?: string;
+      workspaceId: string;
+    }) => LightAgentConfigurationType[];
+    setAgentConfigurationsCache: (
+      workspaceId: string,
+      agentConfigurationsCache: LightAgentConfigurationType[]
+    ) => void;
+    resetAgentConfigurationsCache: (workspaceId: string) => void;
+  };
+}
+
+export interface AppStoreState {
+  userId?: string;
+  workspaceId?: string;
+  timestamp: number | null;
+  agentConfigurationsCache: LightAgentConfigurationType[];
+  actions: {
+    setUserId: (userId: string) => void;
+    getAgentConfigurationsCache: ({
+      userId,
+      workspaceId,
+    }: {
+      userId?: string;
+      workspaceId: string;
+    }) => LightAgentConfigurationType[];
+    setAgentConfigurationsCache: ({
+      userId,
+      workspaceId,
+      agentConfigurationsCache,
+    }: {
+      userId: string;
+      workspaceId: string;
+      agentConfigurationsCache: LightAgentConfigurationType[];
+    }) => void;
+    resetAgentConfigurationsCache: ({
+      userId,
+      workspaceId,
+    }: {
+      userId: string;
+      workspaceId: string;
+    }) => void;
+  };
+}
+
+const indexedDBStorage = {
```

Comment:
I haven't tested it but localStorage is synchronous and will block a main thread, and wonder if it's better to us e indexedDB (but nothing fancy, zustand stringify everything)